### ISSUE TITLE
Initialization of compound_unit

### ISSUE THREAD

**Issue by krofik** [nholthaus/units#268]
I am implementing [this](http://walter.bislins.ch/bloge/index.asp?page=Calculator+for+Refractivity+based+on+Ciddor+Equation) formula. Inside the function I try to do the following

```c++
using micrometer_n2 = inverse<squared<micrometer>>;
static constexpr micrometer_n2 w0{ 295.235 }; // error here
// ... use w0
```

But it does not want to compile due to undefined conversion from double to micrometer_n2. 
Neither does `make_unit<micrometer_n2>(...)` work on initialization. 

I assume the problem is just wrong syntax. Maybe I might use UNIT_ADD macro, but this seems to be way too much work just for local types.

I am using `units.h` version 2.3.1 with MSVC 2017

**Comment by nholthaus** [comment]
I'm not in front of a compiler, but I think what you were intending to do is:

`using micrometer_n2 = decltype(1.0 / (micrometer_t(1) * micrometer_t(1)));`

in version 2.3.1, non-`_t` types refer to the unit conversions, and are probably hard to use without the `UNIT_ADD` macro. I'd suggest that you prefer `decltype` statements for local definitions. Docs definitely need an update.

**Comment by krofik** [comment]
Thanks, this approach seems to work fine, at first. 
I am pretty early in using `units.h` in my project, therefor a switch to version v3.0.0 wouldn't be an issue at all. Has version 3.x.x solved the problem I am attending in 2.3.1?

**Comment by krofik** [comment]
Hello, I have another question about defining custom units: I have a computation which produces [Tesla * meter] = [mass * lenght * time^-2 * current^-1] unit from combination of [ mass * accel * resistance / voltage ]. For convenience and further usage of resulting unit I want to create type alias as follows:

```c++
using tesla_meter_t = unit<std::ratio<1>, compound_unit<magnetic_field_strength::tesla, length::meter>>;
```
But it does not compile with following error message:

```c++
Error C2440 'return': cannot convert from 
'units::unit_t<units::unit<std::ratio<1,1>,units::base_unit<std::ratio<1,1>,std::ratio<1,1>,std::ratio<-2,1>,std::ratio<0,1>,std::ratio<-1,1>,std::ratio<0,1>,std::ratio<0,1>,std::ratio<0,1>,std::ratio<0,1>>,std::ratio<0,1>,std::ratio<0,1>>,double,units::linear_scale>' 
to 
'units::unit<std::ratio<1,1>,units::unit<std::ratio<1,1>,units::base_unit<std::ratio<1,1>,std::ratio<1,1>,std::ratio<-2,1>,std::ratio<0,1>,std::ratio<-1,1>,std::ratio<0,1>,std::ratio<0,1>,std::ratio<0,1>,std::ratio<0,1>>,std::ratio<0,1>,std::ratio<0,1>>,std::ratio<0,1>,std::ratio<0,1>>'
```

The 'proportion' of units seems to be right, but there is clearly mismatch in `units::unit_t`and `units::unit`... How do I fix the definition of `tesla_meter_t` ?
